packages/cli/src/build/utils.ts

```diff
@@ -1,7 +1,40 @@
+import { Project } from "../project";
+import { isTsPath } from "../rollup-plugins/typescript-declarations";
+import { writeDevTSFile } from "../dev";
+import * as fs from "fs-extra";
+import path from "path";
+
 export function getDevPath(cjsPath: string) {
   return cjsPath.replace(/\.js$/, ".dev.js");
 }
 
 export function getProdPath(cjsPath: string) {
   return cjsPath.replace(/\.js$/, ".prod.js");
 }
+
+export async function cleanProjectBeforeBuild(project: Project) {
+  await Promise.all(
+    project.packages.map(async (pkg) => {
+      await Promise.all([
+        fs.remove(path.join(pkg.directory, "dist")),
+        ...pkg.entrypoints
+          .filter((entrypoint) => entrypoint.name !== pkg.name)
```

Comment:
is this filter needed? the piece that I've based my code on didn't do any of this - it just has used `pkg.entrypoints` as the base for computing the paths:
https://github.com/preconstruct/preconstruct/blob/aa54f62e55ce47e6e2c7af59a1ff1f33fbda127e/packages/cli/src/build/rollup.ts#L57-L67

Reply:
That code is doing a different thing. Here we want to clear all the entrypoint dists _and_ the root dist directory because the root dist directory will contain shared chunks and the TS declarations, and the filter is necessary because the root is _sometimes_ an entrypoint so we don't try to delete the same thing at the same time (#306). I suppose it's not technically necessary to remove the root one for watch since there shouldn't be symlinks there if it's not an entrypoint but it's nice for consistency with build and maybe if the root _was_ an entrypoint, preconstruct dev was run, it was removed as an entrypoint and preconstruct watch was run

Reply:
So the summary would be that the root isn't always an entrypoint but the root dist is always created regardless. Good to know - thanks for the explanation 👍

Reply:
> the root dist is always created regardless

it will be created in _some_ cases(currently if someone is using TypeScript or has multiple entrypoints with common modules between them), it won't always be created.